### PR TITLE
new NGE block: Instantiate

### PR DESCRIPTION
**Pull request by deltakosh**
The new instantiate node:
image

I also introduced the notion of LoopID contextual value that let you control the transform per step

cc @Pryme8 @PatrickRyanMS

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14196/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14196/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14196/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14196/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14196/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14196/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14196/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14196/merge#BCU1XR#0

**Comment by PatrickRyanMS** [comment]
@deltakosh, I ran through a bunch of scenarios in my head on this node and I couldn't find one that failed. I think this is good to go.

File: packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts [hunk 1]
```diff
@@ -153,7 +153,9 @@ export class GeometryCollectionBlock extends NodeGeometryBlock {
             if (!availables.length) {
                 return null;
             }
-            return availables[Math.round(Math.random() * (availables.length - 1))];
+            const index = Math.round(Math.random() * (availables.length - 1));
+            console.log(index);
```
Comment (sebavan) [inline packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts:157]: should be removed :-)

**Review by sebavan — APPROVED** [review]
LGTM aside of a small nit but I know nothing of the NGE code so would be good to get @Popov72 eyes as well on it ?

File: packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Meshes/Node/Blocks/geometryCollectionBlock.ts:157]: Gosh right!

**Pull request merged**